demo/site/src/app/layout.tsx

```diff
@@ -13,10 +16,13 @@ export default function RootLayout({
 }: Readonly<{
     children: React.ReactNode;
 }>) {
+    const SitePreviewWrapper = draftMode().isEnabled ? SitePreviewProvider : Fragment;
```

Comment:
Do you think it's better to do this here than in SitePreviewProvider?

Reply:
SitePreviewProvider is (must be) a client component, draftMode() is for server only.

But I could split SitePreviewProvider into to files.

I think it's better that way, as it avoids having a "SitePreviewProvider" wrapper in non-preview/draft mode. Even if it does nothing, it's kind of confusing to have it.

Before app-router we had a custom file for preview paths and only there the SitePreviewProvider was wrapped.

Reply:
> Before app-router we had a custom file for preview paths and only there the SitePreviewProvider was wrapped.

@fraxachun changed that in [#1237](https://github.com/vivid-planet/comet/pull/1237/files#diff-5ae39b382bf2224df93c5f3674fdb1a6c2373ef44372df055c24e8aa2aa4a77a).

Reply:
I'm not sure about the `SitePreviewWrapper` name though. Maybe `ConditionalSitePreview`? Or do it inline in the JSX?

Reply:
> Or do it inline in the JSX?

how?

Reply:
> how?

```tsx
<StyledComponentsRegistry>
  {draftMode().isEnabled ? <SitePreviewProvider>{children}</SitePreviewProvider> : children}
</StyledComponentsRegistry>
```

But maybe that would get more complicated if children become more complex.

Reply:
done